### ISSUE TITLE
Não foi possível reconhecer o formato de imagem do arquivo “/sysroot/home/leandro/.gimp-2.8/patterns/GPS-Pat/Canvas-Covered.xcf”

### ISSUE THREAD

**Issue by LeandroStanger** [flathub/org.gimp.GIMP#3]

**Comment by Jehan** [comment]
Hello!

Thanks for the report, but could you please give an explanation, and in English if possible?
Is this an error message (popup or in the terminal?) which is returned when you try to open an image file? Only for this specific file or any XCF?

If so, could you please upload this XCF somewhere so we can test?

Note: I tried to grep this error message, but I could not find it in our Portuguese translation.

**Comment by LeandroStanger** [comment]
Could not recognize file image format “/sysroot/home/leandro/.gimp-2.8/patterns/GPS-Pat/Canvas-Covered.xcf”

**Comment by Jehan** [comment]
Thanks, but once again, I need to know a bit more than just dropping a message error. ;-)

My assumption so far (but I'd prefer to work with facts, not assumptions) is that you try to open this XCF in the GIMP flatpak installed from Flathub and it fails with this error. Maybe could you upload the XCF file please?

Not sure if you can upload in github. If you can't, upload on some third party and send the link. For instance in: https://framadrop.org/

**Comment by LeandroStanger** [comment]
https://framadrop.org/r/Vs-QvSnLpC#4NKbhuQWsnG6q6y8ZJemyrZPbfFkexhsnCQoD4z7Ncs=

**Comment by Jehan** [comment]
Thanks.

So I just opened this XCF with the flatpak GIMP and it opens fine. No error, nothing.
Could you be more explicit on your issue?

(1) Do you use GIMP from flathub on a Linux machine? If you are not sure, where did you download/install GIMP from?
This repository is only about the flathub release of GIMP.

(2) Where do you see this error?
- Is it on terminal? If so, could you copy-paste the exact error message which you see (in Portuguese is fine)?
- Is it a popup? If so, could you make a screenshot and upload it here?

Thanks.

**Comment by Jehan** [comment]
Also:

(3) What is your operating system?

(4) What version of flatpack are you using?

**Comment by LeandroStanger** [comment]
What is your operating system?
Endless OS 3.3.7
What version of flatpak are you using?
Flatpak 0.10.1

**Comment by LeandroStanger** [comment]
![captura de tela de 2017-12-28 22-45-22](https://user-images.githubusercontent.com/26097419/34426733-d4b3e97e-ec20-11e7-8d37-62ff33357542.png)

**Comment by Jehan** [comment]
Ok if I am not mistaken, Endless has been using flatpak for quite long and therefore may be using a custom flatpak, not the one from flathub.

Could you please run the following commands:

> flatpak remotes -d
> flatpak list -d

And paste here the output of these 2 commands?

Also does this problem happen only with this XCF or with any XCF?
Thanks.

**Comment by LeandroStanger** [comment]
leandro@leandro-stanger:~$ flatpak remotes -d
Nome                   Título                 URL                                          Prioridade Opções           
eos-apps               Endless Applications    https://ostree.endlessm.com/ostree/eos-apps  1          system             
eos-runtimes           Endless OS and Runtimes https://ostree.endlessm.com/ostree/eos-amd64 1          system             
eos-sdk                Endless Apps Runtimes   https://ostree.endlessm.com/ostree/eos-sdk   1          system             
flathub                Flathub                 https://flathub.org/repo/                    1          system             
gnome                  Gnome Stable Runtimes   http://sdk.gnome.org/repo/                   1          system             
gnome_nightly          Nightly GNOME Runtime   https://sdk.gnome.org/nightly/repo/          1          system             
org.gnome.Games-origin Games (Nightly)         https://sdk.gnome.org/nightly/repo-apps/     0          system,no-enumerate
org.gnome.Todo-origin  GNOME To Do             https://sdk.gnome.org/nightly/repo-apps/     0          system,no-enumerate

**Comment by LeandroStanger** [comment]
leandro@leandro-stanger:~$ flatpak list -d
Ref                                                       Origem                 Commit ativo Último commit Tamanho instalado Opções      
com.discordapp.Discord/x86_64/stable                      flathub                9beaca9a3fa3 -               39,5 MB          system,current
com.endlessm.EknServices/x86_64/eos3                      eos-sdk                cb82b33b573a -                1,0 MB          system,current
com.endlessm.animals.pt/x86_64/eos3                       eos-apps               f40bad1a70ad -               66,8 MB          system,current
com.endlessm.astronomy.pt/x86_64/eos3                     eos-apps               5d2551d2a5fe -               43,2 MB          system,current
com.endlessm.biology.pt/x86_64/eos3                       eos-apps               f3169711bc6a -               63,6 MB          system,current
com.endlessm.celebrities.pt/x86_64/eos3                   eos-apps               392b17805c4e -              146,5 MB          system,current
com.endlessm.cooking.pt/x86_64/eos3                       eos-apps               99c2e7279526 -              139,1 MB          system,current
com.endlessm.dinosaurs.pt/x86_64/eos3                     eos-apps               2f660cbb337c -               44,9 MB          system,current
com.endlessm.encyclopedia.pt/x86_64/eos3                  eos-apps               8cb9836b37b1 -                2,6 GB          system,current
com.endlessm.extra.pt_BR/x86_64/eos3                      eos-apps               b3308f941c9d -               66,6 MB          system,current
com.endlessm.farming.pt/x86_64/eos3                       eos-apps               711b44403b5c -               54,9 MB          system,current
com.endlessm.finance/x86_64/eos3                          eos-apps               8586a7cbf0fc -                5,5 MB          system,current
com.endlessm.geography.pt/x86_64/eos3                     eos-apps               28a48325d356 -              218,2 MB          system,current
com.endlessm.history.pt/x86_64/eos3                       eos-apps               aa493c25b2f4 -              165,9 MB          system,current
com.endlessm.howto.pt/x86_64/eos3                         eos-apps               86db4cc1d9bc -              137,2 MB          system,current
com.endlessm.library.pt/x86_64/eos3                       eos-apps               19e04a943d6f -              115,1 MB          system,current
com.endlessm.math.pt/x86_64/eos3                          eos-apps               a989e2da3111 -               25,2 MB          system,current
com.endlessm.myths.pt/x86_64/eos3                         eos-apps               a62c6c6fc0e1 -               38,5 MB          system,current
com.endlessm.omelete.pt/x86_64/eos3                       eos-apps               c084972a1801 -              151,7 MB          system,current
com.endlessm.photos/x86_64/eos3                           eos-apps               5250d277b48f -               64,0 MB          system,current
com.endlessm.physics.pt/x86_64/eos3                       eos-apps               c388026d51ab -               30,5 MB          system,current
com.endlessm.programming/x86_64/eos3                      eos-apps               83cda6895d6e -               14,6 MB          system,current
com.endlessm.programming_guide.pt/x86_64/eos3             eos-apps               493458927311 -               77,3 MB          system,current
com.endlessm.resume/x86_64/eos3                           eos-apps               65ec435a56dc -               12,6 MB          system,current
com.endlessm.soccer.pt/x86_64/eos3                        eos-apps               71d63ff514de -              390,8 MB          system,current
com.endlessm.socialsciences.pt/x86_64/eos3                eos-apps               eb9c9e69e6dc -               37,6 MB          system,current
com.endlessm.translation/x86_64/eos3                      eos-apps               e29ef08d8e5e -                2,1 MB          system,current
com.endlessm.travel.pt/x86_64/eos3                        eos-apps               fbb85af57942 -              330,4 MB          system,current
com.endlessm.videonet/x86_64/eos3                         eos-apps               a4385ab8c3a1 -                3,6 MB          system,current
com.endlessm.weather/x86_64/eos3                          eos-apps               13ee1153eacb -               21,5 MB          system,current
com.endlessm.your_health.pt_BR/x86_64/eos3                eos-apps               79eec960b239 -               37,9 MB          system,current
com.github.Slingshot/x86_64/eos3                          eos-apps               fd2df401bd32 -               26,6 MB          system,current
com.github.wwmm.pulseeffects/x86_64/stable                flathub                0d38bf2aa6a0 -               92,8 MB          system,current
com.github.z.Cumulonimbus/x86_64/stable                   flathub                fd9dc14d13fc -              191,9 MB          system,current
com.google.Chrome/x86_64/eos3                             eos-apps               9b2517dbc2e8 -              602,6 kB          system,current
com.googleplaymusicdesktopplayer.GPMDP/x86_64/stable      flathub                36a855eb62a6 -              201,4 MB          system,current
com.jetbrains.PyCharm-Community/x86_64/stable             flathub                487981acdd6a -              731,4 MB          system,current
com.play0ad.zeroad/x86_64/stable                          flathub                353282652862 -                1,8 GB          system,current
com.scoutshonour.Digital/x86_64/stable                    flathub                ff7b19c70771 -               44,9 MB          system,current
com.skype.Client/x86_64/stable                            flathub                095e4721fd3d -               41,0 MB          system,current
com.spotify.Client/x86_64/stable                          flathub                37fb3d48a9a1 -               48,9 MB          system,current
com.teeworlds.Teeworlds/x86_64/stable                     flathub                7116fc85f8f5 -                9,7 MB          system,current
com.transmissionbt.Transmission/x86_64/eos3               eos-apps               f7b974978d04 -               33,3 MB          system,current
com.uploadedlobster.peek/x86_64/stable                    flathub                57459940c1d0 -                7,3 MB          system,current
com.valvesoftware.Steam/x86_64/stable                     flathub                4819a16e6494 -                6,3 MB          system,current
de.billardgl.Billardgl/x86_64/eos3                        eos-apps               51a7bf57ff79 -                8,0 MB          system,current
io.github.Supertux/x86_64/eos3                            eos-apps               d3653836b491 -               96,5 MB          system,current
io.github.betaflight.BetaflightConfigurator/x86_64/stable flathub                86eee4e9852b -              295,8 MB          system,current
io.lbry.lbry-app/x86_64/stable                            flathub                f04f521d9c3c -              248,2 MB          system,current
io.mgba.mGBA/x86_64/stable                                flathub                bfce60c7272f -                5,8 MB          system,current
net.blockout.Blockout2/x86_64/eos3                        eos-apps               0b0d208fd704 -                5,0 MB          system,current
net.gcompris.Gcompris/x86_64/eos3                         eos-apps               0af79c448f69 -              249,2 MB          system,current
net.mediaarea.BWFMetaEdit/x86_64/stable                   flathub                a22ed09713c9 -                1,2 MB          system,current
net.minetest.Minetest/x86_64/eos3                         eos-apps               0b9b7cf329f7 -               21,8 MB          system,current
net.olofson.Kobodeluxe/x86_64/eos3                        eos-apps               8c80f54d59b8 -                2,1 MB          system,current
net.sourceforge.Atanks/x86_64/eos3                        eos-apps               be25edb1a308 -                9,7 MB          system,current
net.sourceforge.Audacity/x86_64/eos3                      eos-apps               a5f9a53852f2 -               55,0 MB          system,current
net.sourceforge.Btanks/x86_64/eos3                        eos-apps               713d10f74c8f -               36,8 MB          system,current
net.sourceforge.ChromiumBSU/x86_64/eos3                   eos-apps               963f2c867fe9 -                2,6 MB          system,current
net.sourceforge.Extremetuxracer/x86_64/eos3               eos-apps               374f0aeea5b1 -               35,1 MB          system,current
net.sourceforge.Klavaro/x86_64/stable                     flathub                e362c217aa9d -               30,6 MB          system,current
net.sourceforge.Rili/x86_64/eos3                          eos-apps               ecce8675aaa5 -              140,8 MB          system,current
net.sourceforge.Supertuxkart/x86_64/eos3                  eos-apps               933e169a4924 -              315,7 MB          system,current
net.sourceforge.Torcs/x86_64/eos3                         eos-apps               009da6e406de -              458,1 MB          system,current
net.sourceforge.Tuxfootball/x86_64/eos3                   eos-apps               fc1c7d15a00f -               18,1 MB          system,current
net.sourceforge.Warmux/x86_64/eos3                        eos-apps               47e3c612fa0f -              109,2 MB          system,current
net.wz2100.Warzone2100/x86_64/eos3                        eos-apps               94ccb51320bb -               85,8 MB          system,current
org.armagetronad.Armagetronad/x86_64/eos3                 eos-apps               b1503e11fc1a -                3,9 MB          system,current
org.blender.Blender/x86_64/stable                         flathub                f2a609c333b8 -              222,2 MB          system,current
org.debian.Tuxpuck/x86_64/eos3                            eos-apps               2378f3314b12 -              546,3 kB          system,current
org.debian.alioth.tux4kids.Tuxmath/x86_64/eos3            eos-apps               351462a7be06 -               11,3 MB          system,current
org.debian.alioth.tux4kids.Tuxtype/x86_64/eos3            eos-apps               3d50b0acd726 -               11,5 MB          system,current
org.freeciv.Freeciv/x86_64/eos3                           eos-apps               5b3431fa4ef5 -               38,0 MB          system,current
org.frozenbubble.FrozenBubble/x86_64/eos3                 eos-apps               a0d4503a68da -               28,7 MB          system,current
org.gimp.GIMP/x86_64/stable                               flathub                7bda4b1cadf3 -              127,2 MB          system,current
org.gnome.Freecell/x86_64/eos3                            eos-apps               63ec8da82228 -               28,2 MB          system,current
org.gnome.Games/x86_64/master                             org.gnome.Games-origin 80380245430b -               34,1 MB          system,current
org.gnome.Genius/x86_64/eos3                              eos-apps               4b6b1525b4ef -               13,0 MB          system,current
org.gnome.Gnote/x86_64/eos3                               eos-apps               3c228ba5d49c -               11,0 MB          system,current
org.gnome.Iagno/x86_64/eos3                               eos-apps               0cf327988d5c -                8,6 MB          system,current
org.gnome.Quadrapassel/x86_64/eos3                        eos-apps               abb0129307b1 -                6,0 MB          system,current
org.gnome.Solitaire/x86_64/eos3                           eos-apps               f7a9b9a91c09 -               28,2 MB          system,current
org.gnome.Tetravex/x86_64/eos3                            eos-apps               343e449d5d2e -                6,2 MB          system,current
org.gnome.Todo/x86_64/master                              org.gnome.Todo-origin  ad09a6a5d6ac -               13,5 MB          system,current
org.gnome.people.dscorgie.Labyrinth/x86_64/eos3           eos-apps               28d15912cdcd -                5,1 MB          system,current
org.gnucash.GnuCash/x86_64/stable                         flathub                f9348df813e4 -               86,6 MB          system,current
org.inkscape.Inkscape/x86_64/stable                       flathub                490bafd0b8d3 -              150,1 MB          system,current
org.kde.Kalzium/x86_64/eos3                               eos-apps               6e09090a961b -               34,2 MB          system,current
org.kde.Kapman/x86_64/eos3                                eos-apps               98ef8baad086 -                2,1 MB          system,current
org.kde.Katomic/x86_64/eos3                               eos-apps               e047ca637e14 -              944,6 kB          system,current
org.kde.Kblocks/x86_64/eos3                               eos-apps               8d8d49c9f497 -                3,4 MB          system,current
org.kde.Kbounce/x86_64/eos3                               eos-apps               f58f28036d2d -                4,7 MB          system,current
org.kde.Kbruch/x86_64/eos3                                eos-apps               e75e916dec48 -                1,3 MB          system,current
org.kde.Kdiamond/x86_64/eos3                              eos-apps               3a1b5710233f -                4,4 MB          system,current
org.kde.Kgeography/x86_64/eos3                            eos-apps               dfb52cea39f4 -                7,4 MB          system,current
org.kde.Kgoldrunner/x86_64/eos3                           eos-apps               2f62cb64fcd8 -                5,4 MB          system,current
org.kde.Khangman/x86_64/eos3                              eos-apps               a65c7f3a2b55 -                3,0 MB          system,current
org.kde.Kigo/x86_64/eos3                                  eos-apps               4f8c1884768b -               10,1 MB          system,current
org.kde.Killbots/x86_64/eos3                              eos-apps               7bf6d88964f9 -                1,3 MB          system,current
org.kde.Kjumpingcube/x86_64/eos3                          eos-apps               0e91ba3f551e -              490,5 kB          system,current
org.kde.Klines/x86_64/eos3                                eos-apps               c7520136fb7b -                6,6 MB          system,current
org.kde.Knavalbattle/x86_64/eos3                          eos-apps               b69e09d40688 -                1,3 MB          system,current
org.kde.Knetwalk/x86_64/eos3                              eos-apps               b5aafad63150 -                1,0 MB          system,current
org.kde.Ksame/x86_64/eos3                                 eos-apps               bebe7fdb71ae -                2,0 MB          system,current
org.kde.Ksquares/x86_64/eos3                              eos-apps               f339dd7a36e5 -              274,9 kB          system,current
org.kde.Ksudoku/x86_64/eos3                               eos-apps               f79fef8c2993 -                3,1 MB          system,current
org.kde.Ktuberling/x86_64/eos3                            eos-apps               ffbf8e9876d5 -                7,7 MB          system,current
org.kde.Kubrick/x86_64/eos3                               eos-apps               116ceeb80810 -              419,8 kB          system,current
org.kde.Kwordquiz/x86_64/eos3                             eos-apps               b78d36ff3ca6 -                1,9 MB          system,current
org.kde.Palapeli/x86_64/eos3                              eos-apps               a2728d7113fe -                2,5 MB          system,current
org.kicad_pcb.KiCad/x86_64/stable                         flathub                fd929b9e017a -                3,0 GB          system,current
org.maemo.Numptyphysics/x86_64/eos3                       eos-apps               1ba89a0f7ac6 -              624,1 kB          system,current
org.marsshooter.Marsshooter/x86_64/eos3                   eos-apps               dcb07fd62548 -               71,3 MB          system,current
org.megaglest.MegaGlest/x86_64/stable                     flathub                6c6e3deebab6 -              502,7 MB          system,current
org.mozilla.Firefox/x86_64/eos3                           eos-apps               71cf707fdebb -              644,1 kB          system,current
org.mypaint.MyPaint/x86_64/stable                         flathub                167d5ca554c3 -               61,2 MB          system,current
org.openarena.Openarena/x86_64/eos3                       eos-apps               85d074f0e09c -              438,7 MB          system,current
org.openscad.Openscad/x86_64/eos3                         eos-apps               a60c2643099e -                6,6 MB          system,current
org.pitivi.Pitivi/x86_64/eos3                             eos-apps               f0b692b33ea6 -               25,1 MB          system,current
org.seul.Pingus/x86_64/eos3                               eos-apps               08e8a3929bca -               24,6 MB          system,current
org.squeakland.Etoys/x86_64/eos3                          eos-apps               29563aa1d4a1 -               43,3 MB          system,current
org.squeakland.Scratch/x86_64/eos3                        eos-apps               91dbb1347170 -               48,5 MB          system,current
org.stellarium.Stellarium/x86_64/eos3                     eos-apps               02d15baae1e0 -               81,3 MB          system,current
org.sugarlabs.Turtleblocks/x86_64/eos3                    eos-apps               68ba5e69fba5 -               12,1 MB          system,current
org.telegram.desktop/x86_64/stable                        flathub                bfa7d1d01929 -               40,3 MB          system,current
org.tordini.flavio.Minitube/x86_64/stable                 flathub                7abfaf25d2ab -               10,4 MB          system,current
org.tuxfamily.Xmoto/x86_64/eos3                           eos-apps               e8554e481991 -               71,9 MB          system,current
org.tuxpaint.Tuxpaint/x86_64/eos3                         eos-apps               c2d23d48782d -                6,7 MB          system,current
org.videolan.VLC/x86_64/eos3                              eos-apps               286cb4f0aa85 -              212,3 MB          system,current
org.wesnoth.Wesnoth/x86_64/eos3                           eos-apps               876264fc538a -              481,1 MB          system,current
com.endlessm.Platform/x86_64/eos3.0                       eos-runtimes           02a2ea161f20 -                2,0 GB          system,runtime
com.endlessm.Platform/x86_64/eos3.1                       eos-runtimes           0894b7c249b4 -                2,2 GB          system,runtime
com.endlessm.Platform/x86_64/eos3.2                       eos-runtimes           46059fc0a5b8 -                2,2 GB          system,runtime
com.endlessm.apps.Platform/x86_64/1                       eos-sdk                9e248b907d60 -              883,0 MB          system,runtime
org.freedesktop.Platform.Compat32/x86_64/1.6              flathub                063b998c53a0 -              578,4 MB          system,runtime
org.freedesktop.Platform.VAAPI.Intel/i386/1.6             flathub                e39a844199ec -                7,2 MB          system,runtime
org.freedesktop.Platform.VAAPI.Intel/x86_64/1.6           flathub                5962b0bb8996 -                7,8 MB          system,runtime
org.freedesktop.Platform/i386/1.6                         flathub                e4f753e5302d -              578,4 MB          system,runtime
org.freedesktop.Platform/x86_64/1.6                       flathub                efe510102888 -              562,2 MB          system,runtime
org.freedesktop.Sdk.Extension.gfortran-62/x86_64/1.6      flathub                5b89f0da9d35 -              341,9 MB          system,runtime
org.freedesktop.Sdk/x86_64/1.6                            flathub                058ebf0741fe -                1,3 GB          system,runtime
org.gnome.Games.Locale/x86_64/stable                      flathub                ca22496b7857 -                3,2 MB          system,runtime
org.gnome.Platform/x86_64/3.24                            gnome                  02c1518cd9a9 -              734,2 MB          system,runtime
org.gnome.Platform/x86_64/3.26                            flathub                bfd4b1950361 -              777,1 MB          system,runtime
org.gnome.Platform/x86_64/master                          gnome_nightly          860ae022c4e0 -              784,3 MB          system,runtime
org.kde.Platform/x86_64/5.9                               flathub                dfd2fafe16ab -                1,1 GB          system,runtime

**Comment by Jehan** [comment]
It looks like you indeed have the flathub GIMP with the last commit. But here it works fine. I saw that you were using org.gnome.Platform/x86_64/3.24 from gnome though I was using the one from flathub so I uninstalled and reinstalled it from gnome. Now I have the same versions of org.gnome.Platform/x86_64/3.24 and org.gimp.GIMP/x86_64/stable (same commit hashes). Yet it still works perfectly here when I open this XCF.

So I am a little lost because flatpak is all about reproducible builds, which are meant to be working the same everywhere, etc.

@alexlarsson What could be different for the same flatpak GIMP using the same org.gnome.platform (all at the same commit hash) to work differently on different computers?

P.S.: for info, after some code research, the outputted error is coming from GdkPixbuf code, file `gdk-pixbuf/gdk-pixbuf/gdk-pixbuf-io.c`, function `_gdk_pixbuf_get_module()` and the original (English) error is: "Couldn't recognize the image file format for file '%s'".
